tests/test_provider_azure.py

```diff
@@ -4152,6 +4200,23 @@ def test_ensure_required_root_ns_values(self):
         # same object, no copy
         self.assertEqual(id(zone), id(ret))
 
+    def test_cli(self):
+        self._get_cli_provider()
```

Comment:
This doesn't appear to actually test anything. It's just creating some mocks and a provider with cli enabled.

`_get_provider` is a helper method that's being called by other `def test_*` methods which go on to test things. This is just calling a helper style method and not doing anything with the results. 

Probably makes sense to ditch this method, turn `_get_cli_provider` into `test_cli_provider` and do a bit of checking in there to make sure that things are configured correctly and that a `AzureCliCredential` is in place as expected.

Reply:
I _think_ I fixed it

Reply:
Doesn't look like any changes were pushed up after this comment.

Reply:
[8feb496](https://github.com/octodns/octodns-azure/pull/84/commits/8feb496c8c2529cdfba67e6a1b4707f88d5aaad8) is in this PR and it looks like tests pass

Reply:
As mentioned above the results after that PR calls the code with the new parameter but there are no actual tests to see if it's doing what it's supposed to. No `self.assert*`

Reply:
I wonder how we can test anything without actually installing the azure CLI and having a domain to test against? This function just returns the client object. I am open to suggestions on how to properly test this.

Reply:
My first comment in this thread has my thoughts on the changes that need to be made and the tests that would apply:

> Probably makes sense to ditch this method, turn _get_cli_provider into test_cli_provider and do a bit of checking in there to make sure that things are configured correctly and that a AzureCliCredential is in place as expected.

The configured correctly bit i referring to checking `self._client_method`'s value. and the `AzureCliCredential` would likely be a `self.assertIsInstance`